packages/eslint-plugin/lib/rules/index.js

```diff
@@ -103,3 +103,8 @@ module.exports = {
   // export new rule here ↑
   // DO NOT REMOVE THIS COMMENT
 };
+
+module.exports =
+  /** @type {Record<string, import('@eslint/core').RuleDefinition>} */ (
+    /** @type {unknown} */ (rules)
+  );
```

Comment:
The rules are based on the ESLint v8 types, supportinh the ESLint v9 types (specifically the language types) was a much bigger endeavour.
I figured that if it wasn't broke, don't fix (it in the same PR).